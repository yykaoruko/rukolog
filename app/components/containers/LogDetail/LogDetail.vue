<template>
  <article class="log-detail">
    <h1 class="log-detail__title">{{ title }}</h1>
    <app-markdown class="log-detail__body" :body="body" />
  </article>
</template>

<script lang="ts">
import Vue from 'vue';
import AppDate from '@/components/basics/AppDate/AppDate.vue';
import AppMarkdown from '@/components/basics/AppMarkdown/AppMarkdown.vue';
export default Vue.extend({
  name: 'LogDetail',
  components: {
    AppMarkdown,
  },
  props: {
    title: {
      type: String,
      required: true,
    },
    body: {
      type: String,
      required: true,
    },
  },
});
</script>

<style lang="scss" scoped>
.log-detail {
  color: inherit;
}
</style>
