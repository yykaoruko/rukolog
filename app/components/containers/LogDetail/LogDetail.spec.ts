import Vue from 'vue';
import { shallowMount, Wrapper } from '@vue/test-utils';
import LogDetail from './LogDetail.vue';

describe('LogDetail', () => {
  describe('mount', () => {
    let wrapper: Wrapper<Vue>;

    beforeEach(() => {
      wrapper = shallowMount(LogDetail, {
        propsData: {
          body: 'body',
          title: 'title',
        },
      });
    });

    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true);
    });

    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
