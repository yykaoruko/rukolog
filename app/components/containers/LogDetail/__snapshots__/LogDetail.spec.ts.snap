// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogDetail mount snapshot 1`] = `
<article
  class="log-detail"
>
  <h1
    class="log-detail__title"
  >
    title
  </h1>
   
  <app-markdown-stub
    body="body"
    class="log-detail__body"
  />
</article>
`;
